src/selector/support.js

```diff
@@ -1,27 +1,20 @@
+import document from "../var/document.js";
 import support from "../var/support.js";
 
-// Support: IE 11+
-// IE doesn't support `CSS.supports( "selector(...)" )`; it will throw
-// in this support test.
+// Support: Chrome 105 - 110+, Safari 15.4 - 16.3+
+// Make sure forgiving mode is not used in `:has()`.
+// `*` is needed as Safari & newer Chrome implemented something in between
+// for `:has()` - it throws in `qSA` if it only contains an unsupported
+// argument but multiple ones, one of which is supported, are fine.
+// We want to play safe in case `:is()` gets the same treatment.
+//
+// Note that we don't need to detect the complete lack of support for `:has()`
+// as then the `qSA` path will throw and fall back to jQuery traversal anyway.
```

Comment:
```suggestion
// Make sure the the `:has()` argument is parsed unforgivingly.
// We include `*` in the test to detect buggy implementations that are
// _selectively_ forgiving (specifically when the list includes at least
// one valid selector).
// Note that we treat complete lack of support for `:has()` as if it were
// spec-compliant support, which is fine because use of `:has()` in such
// environments will fail in the qSA path and fall back to jQuery traversal
// anyway.
```